src/test/java/seedu/address/logic/parser/book/AddBookCommandParserTest.java

```diff
@@ -0,0 +1,126 @@
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.book.AddBookCommand;
+import seedu.address.model.book.Book;
+import seedu.address.model.book.BookName;
+import seedu.address.model.book.Isbn;
+import seedu.address.model.tag.Tag;
+import seedu.address.testutil.BookBuilder;
+
+public class AddBookCommandParserTest {
+    private AddBookCommandParser parser = new AddBookCommandParser();
+
+    @Test
+    public void parse_allFieldsPresent_success() {
+        Book expectedBook = new BookBuilder(HARRY_POTTER).withTags(VALID_TAG_MAGIC).build();
+
+        // whitespace only preamble
+        assertParseSuccess(parser, PREAMBLE_WHITESPACE + BOOK_NAME_DESC_HARRY_POTTER + ISBN_DESC_HARRY_POTTER
+                + AUTHOR_DESC_JK_ROWLING + TAG_DESC_MAGIC, new AddBookCommand(expectedBook));
+
+        // multiple names - last name accepted
+        assertParseSuccess(parser, BOOK_NAME_DESC_HUNGER_GAMES + BOOK_NAME_DESC_HARRY_POTTER
+                + ISBN_DESC_HARRY_POTTER + AUTHOR_DESC_JK_ROWLING
+                + TAG_DESC_MAGIC, new AddBookCommand(expectedBook));
+
+        // multiple isbn - last isbn accepted
+        assertParseSuccess(parser, BOOK_NAME_DESC_HARRY_POTTER + ISBN_DESC_HUNGER_GAMES + ISBN_DESC_HARRY_POTTER
+                + AUTHOR_DESC_JK_ROWLING + TAG_DESC_MAGIC, new AddBookCommand(expectedBook));
+
+        // multiple tags - all accepted
+        Book expectedBookMultipleTags = new BookBuilder(HARRY_POTTER).withTags(VALID_TAG_ADVENTURE, VALID_TAG_MAGIC)
+                .build();
+        assertParseSuccess(parser, BOOK_NAME_DESC_HARRY_POTTER + ISBN_DESC_HARRY_POTTER
+                + AUTHOR_DESC_JK_ROWLING
+                + TAG_DESC_ADVENTURE + TAG_DESC_MAGIC, new AddBookCommand(expectedBookMultipleTags));
+
+        // multiple authors - all accepted
+        Book expectedBookMultipleAuthors = new BookBuilder(HARRY_POTTER)
+                .withAuthors(VALID_AUTHOR_JK_ROWLING, VALID_AUTHOR_SUZANNE_COLLINS)
+                .withTags()
+                .build();
+        assertParseSuccess(parser, BOOK_NAME_DESC_HARRY_POTTER + ISBN_DESC_HARRY_POTTER
+                + AUTHOR_DESC_JK_ROWLING + AUTHOR_DESC_SUZANNE_COLLINS,
+                new AddBookCommand(expectedBookMultipleAuthors));
+    }
+
+    @Test
+    public void parse_optionalFieldsMissing_success() {
+        // zero tags
+        Book expectedBook = new BookBuilder(HARRY_POTTER).withTags().build();
+        assertParseSuccess(parser, BOOK_NAME_DESC_HARRY_POTTER + ISBN_DESC_HARRY_POTTER
+                + AUTHOR_DESC_JK_ROWLING,
+                new AddBookCommand(expectedBook));
+
+        // zero authors
+        expectedBook = new BookBuilder(HARRY_POTTER).withAuthors().build();
+        assertParseSuccess(parser, BOOK_NAME_DESC_HARRY_POTTER + ISBN_DESC_HARRY_POTTER
+                        + TAG_DESC_ADVENTURE + TAG_DESC_MAGIC,
+                new AddBookCommand(expectedBook));
+    }
```

Comment:
Good test for zero tags and zero authors which are optional arguments for the book add command!